### PR TITLE
[air] Consolidate ray.ml.train and ray.ml.predictors

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

Trainer implementations usually come with predictors. It feels odd to keep them in separate modules - instead, we should ship them together in a consolidated `integrations` module.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-air/getting-started.rst [hunk 1]
```diff
@@ -44,31 +44,44 @@ Trainer
...
     :members:
     :show-inheritance:
 
-.. autoclass:: ray.ml.train.data_parallel_trainer.DataParallelTrainer
+.. autoclass:: ray.ml.integrations.data_parallel_trainer.DataParallelTrainer
     :members:
     :show-inheritance:
 
-.. autoclass:: ray.ml.train.gbdt_trainer.GBDTTrainer
+.. autoclass:: ray.ml.integrations.gbdt_trainer.GBDTTrainer
```
Comment (matthewdeng) [inline doc/source/ray-air/getting-started.rst:84]: Hmm are these considered integrations?

Comment (krfricke) [reply doc/source/ray-air/getting-started.rst:84]: GBDT - yes, they are the base class for xgboost and lightgbm, both integrations.
DP - less clear hear imo, but again base class for mostly integrations. I would currently vote to keep them there so they are easy to find. The alternative would be to have `ray.ml.base` or `ray.ml.base_trainers` or so. 
What do other people think?

File: doc/source/ray-air/getting-started.rst [hunk 2]
```diff
@@ -44,31 +44,44 @@ Trainer
...
 
-.. automodule:: ray.ml.train.integrations.xgboost
+
+Predictors
+~~~~~~~~~~
+
+.. autoclass:: ray.ml.predictor.Predictor
+    :members:
+
+.. autoclass:: ray.ml.batch_predictor.BatchPredictor
```
Comment (gjoliver) [inline doc/source/ray-air/getting-started.rst:54]: why are predictor and batch_predictor in 2 separate models?

Comment (krfricke) [reply doc/source/ray-air/getting-started.rst:54]: I think mostly to separate the interface from an implementation. It was in the same module in a prior iteration of the original PR, cc @ericl who suggested the separation

Comment (ericl) [reply doc/source/ray-air/getting-started.rst:54]: I think these are separate enough in use cases two modules adds to clarity.

File: doc/source/ray-air/getting-started.rst [hunk 3]
```diff
@@ -44,31 +44,44 @@ Trainer
...
     :members:
     :show-inheritance:
 
-.. automodule:: ray.ml.train.integrations.sklearn
+.. automodule:: ray.ml.integrations.sklearn
     :members:
     :show-inheritance:
 
-.. autoclass:: ray.ml.train.data_parallel_trainer.DataParallelTrainer
+.. autoclass:: ray.ml.integrations.data_parallel_trainer.DataParallelTrainer
```
Comment (ericl) [inline doc/source/ray-air/getting-started.rst:80]: This shouldn't be an integration right? More like a base class / template.

Comment (ericl) [reply doc/source/ray-air/getting-started.rst:80]: How about `ray.ml.utils.DataParallelTrainer`?

**Comment by amogkam** [comment]
Did't we decide that the end state should be to move all the trainers to ray.train?

In that case, I think it makes more sense to leave the trainers under ray.ml.train and then move them over in one go later, instead of changing around our package structure multiple times.

**Comment by krfricke** [comment]
Superseded by #25449

**Pull request closed without merging**